File: 062.js
//  062
//  Unique Paths
//  2015-12-16
/////////////////////////////////////////////////////
/**
 * @param {number} m
 * @param {number} n
 * @return {number}
 */
var uniquePaths = function(m, n) {
    var t = [];
    for (var i = 0; i < n; i++) {
        t.push(1);
    }
    for (i = 1; i < m; i++) {
        for (var j = 1; j < n; j++) {
            t[j] += t[j-1];
        }
    }
    return t[n-1];
};

/////////////////////////////////////////////////////
var m = 3;
var n = 7;
var r = uniquePaths(m, n);
console.log(r);

File: 062.rb
# 062
# Unique Paths
# 2015-12-16
#####################################################
# @param {Integer} m
# @param {Integer} n
# @return {Integer}
def unique_paths(m, n)
  t = [1]*n
  (m-1).times do
    1.upto(n-1) do |i|
      t[i] += t[i-1]
    end
  end
  t[n-1]
end

#####################################################
m = 3
n = 7
r = unique_paths(m, n)
puts r

